sim/midas/src/main/scala/midas/platform/ZynqShim.scala

```diff
@@ -7,23 +7,26 @@ import chisel3._
 import chisel3.util._
 import junctions._
 import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.diplomacy.{LazyModule}
 import freechips.rocketchip.util.ParameterizedBundle
 
-import midas.core.{HostMemChannelNastiKey, ChannelWidth}
+import midas.core.ChannelWidth
 
-abstract class PlatformShim(implicit p: Parameters) extends Module {
-  def top: midas.core.FPGATop
+abstract class PlatformShim(implicit p: Parameters) extends MultiIOModule {
+  def top: midas.core.FPGATopImp
   def headerConsts: Seq[(String, Long)]
   def genHeader(sb: StringBuilder, target: String) {
     import widgets.CppGenerationUtils._
+    //We need to split out a C++ header from the C header or VCS croaks
+    //sb.append("#include <cstdint>\n")
```

Comment:
Does this comment need to be here anymore?

Reply:
This is related to your question about the `unsigned long long` change. As it turns out, we were never emitting `uint64_t`, if we had been we would have run into compilation errors in VCS ML sims. Since we include this header in basically every compilation unit (= bug), VCS 
compiles it part of C and C++ files and complains about the library we use to include the fixed-width types.

Obviously, we should emit fixed-width types. The easiest way to do this IMO is split the header files, into a const.h that is included in all of the places the legacy midas code wants it, and a const.hpp which just includes all the bridge good stuff we use.

Reply:
I'm pretty sure if you use `#include <stdint.h>` it should work in C or C++.